app/api/custom/orders.py

```diff
@@ -0,0 +1,93 @@
+from flask import Blueprint, jsonify, request
+from flask_jwt_extended import current_user, jwt_required
+from sqlalchemy.orm.exc import NoResultFound
+from flask_limiter.util import get_remote_address
+
+
+from app import limiter
+from app.models import db
+from app.api.auth import return_file
+from app.api.helpers.db import safe_query
+from app.api.helpers.mail import send_email_to_attendees
+from app.api.helpers.errors import ForbiddenError, NotFoundError, UnprocessableEntityError
+from app.api.helpers.order import create_pdf_tickets_for_holder, calculate_order_amount
+from app.api.helpers.storage import UPLOAD_PATHS
+from app.api.helpers.storage import generate_hash
+from app.api.helpers.ticketing import TicketingManager
+from app.api.helpers.permission_manager import has_access
+from app.models.discount_code import DiscountCode
+from app.models.order import Order
+
+order_blueprint = Blueprint('order_blueprint', __name__, url_prefix='/v1/orders')
+
+
+@order_blueprint.route('/invoices/<string:order_identifier>')
+@jwt_required
+def order_invoices(order_identifier):
+    if current_user:
+        try:
+            order = Order.query.filter_by(identifier=order_identifier).first()
+        except NoResultFound:
+            return NotFoundError({'source': ''}, 'Order Invoice not found').respond()
+        if current_user.can_download_tickets(order):
+            key = UPLOAD_PATHS['pdf']['order'].format(identifier=order_identifier)
+            file_path = '../generated/invoices/{}/{}/'.format(key, generate_hash(key)) + order_identifier + '.pdf'
+            try:
+                return return_file('invoice', file_path, order_identifier)
+            except FileNotFoundError:
+                create_pdf_tickets_for_holder(order)
+                return return_file('invoice', file_path, order_identifier)
+        else:
+            return ForbiddenError({'source': ''}, 'Unauthorized Access').respond()
+    else:
+        return ForbiddenError({'source': ''}, 'Authentication Required to access Invoice').respond()
```

Comment:
This function and the event invoice function should be present in 1 file - invoices.py